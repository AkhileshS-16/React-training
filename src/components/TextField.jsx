import { forwardRef } from "react";
import React from "react";
/* eslint-disable react/prop-types */
const TextField = forwardRef(function Textfield(props, ref) {
  const onChange = (e) => {
    if (props.handleSubmit) props.handleSubmit(e.target.value);
  };

  // useEffect(() => {
  //   if (text.length > 5) alert("Length>5");
  //   console.log(text);
  // }, [text]);

  return (
    <span>
      <label htmlFor={props.id}>{props.label}</label>
      <input
        type={props.type}
        data-testid="texttest"
        className="logininput"
        id={props.id}
        placeholder={props.label}
        value={props.value}
        onChange={onChange}
        ref={ref}
      />
      {props.error ? <div>{props.error}</div> : ""}
    </span>
  );
});
export default TextField;
