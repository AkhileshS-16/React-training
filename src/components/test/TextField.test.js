import React from "react";
import TextField from "../TextField";
import { fireEvent, render } from "@testing-library/react";

describe("Test TextField", () => {
  test("TextField Render", () => {
    const { getByTestId } = render(<TextField />);
    getByTestId("texttest");
  });
  test("Check onChange", () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<TextField handleSubmit={onChange} />);
    const element = getByTestId("texttest");
    fireEvent.change(element, { target: { value: "test" } });
    expect(onChange).toHaveBeenCalled();
    expect(element.value).toBe("test");
  });
  test("Check props", () => {
    const value = "test";
    const label = "test";
    const id = "test";
    const { getByTestId } = render(
      <TextField value={value} id={id} placeholder={label} />
    );
    const element = getByTestId("texttest");
    expect(element.value).toBe(value);
    expect(element.id).toBe(id);
  });
  test("check snapshot", () => {
    const value = "test";
    const label = "test";
    const id = "test";
    const { getByTestId } = render(
      <TextField value={value} id={id} label={label} />
    );
    const element = getByTestId("texttest");
    expect(element).toMatchSnapshot();
  });
});
