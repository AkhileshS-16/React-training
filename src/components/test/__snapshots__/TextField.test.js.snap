// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextField check snapshot 1`] = `
<input
  class="logininput"
  data-testid="texttest"
  id="test"
  placeholder="test"
  value="test"
/>
`;
